pkg/accprovisioninput/apis/aci.ctrl/v1alpha1/types.go

```diff
@@ -0,0 +1,126 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AccProvisionInput is a specification for a AccProvisionInput resource
+type AccProvisionInput struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   AccProvisionInputSpec   `json:"spec,omitempty"`
+	Status AccProvisionInputStatus `json:"status,omitempty"`
+}
+
+// AccProvisionInputSpec defines the desired state of AccProvisionInput
+type AccProvisionInputSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "acioperator-sdk generate k8s" to regenerate code after modifying this file
+	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
+	Flavor            string           `json:"flavor"`
+	Config            string           `json:"config"`
+	AccProvisionInput AccProvisionSpec `json:"acc_provision_input"`
+}
+
+type AccProvisionSpec struct {
+	AciConfig     AciConfigSpec     `json:"aci_config,omitempty"`
+	NetConfig     NetConfigSpec     `json:"net_config,omitempty"`
+	Registry      RegistrySpec      `json:"registry,omitempty"`
+	Logging       LoggingSpec       `json:"logging,omitempty"`
+	IstioConfig   IstioConfigSpec   `json:"istio_config,omitempty"`
+	DropLogConfig DropLogConfigSpec `json:"drop_log_config,omitempty"`
+	Multus        MultusSpec        `json:"multus,omitempty"`
+	KubeConfig    KubeConfigSpec    `json:"kube_config,omitempty"`
+}
+
+type AciConfigSpec struct {
+	ClusterTenant string         `json:"cluster_tenant,omitempty"`
+	SyncLogin     *SyncLoginSpec `json:"sync_login,omitempty"`
+	ClientSSL     bool           `json:"client_ssl,omitempty"`
+}
+
+type SyncLoginSpec struct {
+	CertFile string `json:"certfile,omitempty"`
+	KeyFile  string `json:"keyfile,omitempty"`
+}
+
+type NetConfigSpec struct {
+	InterfaceMTU       int  `json:"interface_mtu,omitempty"`
+	SvcMonitorInterval int  `json:"service_monitor_interval,omitempty"`
+	PBRTrackingNonSNAT bool `json:"pbr_tracking_non_snat,omitempty"`
+	PodSubnetChunkSize int  `json:"pod_subnet_chunk_size,omitempty"`
+}
+
+type RegistrySpec struct {
+	ImagePrefix                    string `json:"image_prefix,omitempty"`
+	ImagePullSecret                string `json:"image_pull_secret,omitempty"`
+	AccProvisionOperatorVersion    string `json:"acc_provision_operator_version,omitempty"`
+	AciContainersControllerVersion string `json:"aci_containers_controller_version,omitempty"`
+	AciContainersHostVersion       string `json:"aci_containers_host_version,omitempty"`
```

Comment:
aci_containers_operator_version is missing here...

Reply:
The new Ansible operator generates the acicnioperator CR spec which currently includes the CNI deployment file (https://github.com/noironetworks/acc-provision/blob/master/provision/acc_provision/templates/aci-containers.yaml) but not the acicnioperator deployment spec. Do we want the Ansible operator to manage the acicnioperator as well?

Reply:
I am not sure I understand the question, but we have one "operator" which should be able to manage all the components of the ACI CNI. So, when the image tag for the acc-provision-operator container is updated, it should result in all the components getting updated accordingly.

Also, independently, it should be possible to only update the aci-containers-operator container image tag (as can be done with other ACI CNI images listed here).